lib/ansible/modules/system/macos_pref.py

```diff
@@ -0,0 +1,1079 @@
+            - See U(https://developer.apple.com/library/content/documentation/CoreFoundation/Conceptual/CFPreferences/Concepts/PreferenceDomains.html)
+              for further information.
+        required: false
+        default: currentUser
+        choices: ["anyUser", "currentUser"]
+    host:
+        description:
+            - The host on which the preference should apply. Most Apple preferences
+              live in the C(anyHost) domain.
+            - See U(https://developer.apple.com/library/content/documentation/CoreFoundation/Conceptual/CFPreferences/Concepts/PreferenceDomains.html)
+              for further information.
+        required: false
+        default: anyHost
+        choices: ["anyHost", "currentHost"]
+    key:
+        description:
+            - The key of the preference. Nested values can be accessed by giving all
+              keys and indexes separated by colons (:). Indexes are zero-based
+              (see Notes for special cases).
+        required: true
+    type:
+        description:
+            - The type of the value to write. If unspecified, I(type) will
+              be deduced mostly the same way YAML casts types (see Notes for special cases).
+        required: false
+        default: null
+        choices: ["array", "bool", "boolean", "data", "date", "dict", "float", "real", "int", "integer", "string"]
+    value:
+        description:
+            - The value that will be set for the specified key.
+              If C(value) is omitted and C(state) is not C(absent), the preference value of C(key) will be returned.
+        required: false
+    state:
+        description:
+            - The state of the preference.
+              c(merge) performs a deep merge of dictionaries and arrays.
+        required: false
+        default: replace
+        choices: ["replace", "merge", "absent"]
+notes:
+    - macOS caches preferences aggressively. This module should take care of
+      updating caches but in some cases you may need to logout and login to
+      apply the changes.
+    - Check mode can be use with this module.
+    - Nested keys need to be quoted when use in abbreviated form
+      (because they contains colons and that messes with the syntax).
+    - Dates need to be quoted.
+    - First level (not nested) quoted boolean, integers and floats are
+      converted to boolean, integer and float unless C(type) is specified.
+    - Lists and dicts can't be given as C(value) in C(key=argument) form.
+    - Binary data needs to be encoded in base64.
+'''
+
+EXAMPLES = '''
+# Set a string key
+- name: Set preferred view style
+  macos_pref:
+    domain: com.apple.finder
+    key: FXPreferredViewStyle
+    value: Nlsv
+
+
+# Read a string key
+- name: Read preferred view style
+  macos_pref:
+    domain: com.apple.finder
+    key: FXPreferredViewStyle
+  register: finder_view_style
+
+- name: Check preferred view style
```

Comment:
We don't tend to add checks in the examples, but if we do it's probably better to use the 'assert' module.